app/jobs/solid_cable/trim_job.rb

```diff
@@ -1,9 +1,9 @@
 # frozen_string_literal: true
 
 module SolidCable
-  class PruneJob < ActiveJob::Base
+  class TrimJob < ActiveJob::Base
     def perform
-      ::SolidCable::Message.prunable.delete_all
+      ::SolidCable::Message.trimmable.delete_all
```

Comment:
While this should work pretty well with SQLite, I have some worries about how this would behave on MySQL or PostgreSQL.

It's deleting an unbounded number of messages so could lock for a fair amount of time. If the database is being replicated, that could also trigger replication lag as those deletes are processed.

Also there could generally be locking issues with concurrent jobs attempting to run the query.

The approach solid_cache takes is to delete small amounts of data but do it often.

- Every N / 2 writes we trigger an expiration task (a job or just in a thread).
- The task will try to expire up to N records.

We expire N records, but trigger the expiration after N/2 inserts so we have downward pressure on the cache size when it is too large. But we don't try to clear everything out at once as that could be millions and millions of records.

Solid Cache then has a slightly complicated process for deleting records in a concurrent safe manner, but I think we could maybe just rely on `SKIP LOCKED` here instead. That means you need MySQL 8.0 at least, but Solid Queue already requires that so I don't think it would be an issue to have Solid Cable do the same.

Reply:
Put up https://github.com/npezza93/solid_cable/pull/15 which should address this. Let me know what you think!